modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java

```diff
@@ -59,7 +62,14 @@ public String getDriverClassName() {
     @Override
     public String getJdbcUrl() {
         // Disable Postgres driver use of java.util.logging to reduce noise at startup time
-        return "jdbc:postgresql://" + getContainerIpAddress() + ":" + getMappedPort(POSTGRESQL_PORT) + "/" + databaseName + "?loggerLevel=OFF";
+        return format("jdbc:postgresql://%s:%d/%s?loggerLevel=OFF", getContainerIpAddress(), getMappedPort(POSTGRESQL_PORT), databaseName);
+    }
+
+    @Override
+    protected String constructUrlForConnection(String queryString) {
+        return "".equals(queryString)
+            ? getJdbcUrl()
+            : getJdbcUrl() + QUERY_PARAM_SEPARATOR + queryString.substring(1);
```

Comment:
I believe this will break (would be nice to add such test, btw) if `getJdbcUrl()` returns `jdbc:postgresql://host:port/database` (e.g. a class that overrides `getJdbcUrl()`)